addons/binding/org.openhab.binding.russoundserial/src/main/java/org/openhab/binding/russoundserial/internal/RussoundSerialHandlerFactory.java

```diff
@@ -0,0 +1,89 @@
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import com.thejholmes.russound.Russound;
+import com.thejholmes.russound.RussoundCommandSender;
+import com.thejholmes.russound.RussoundCommander;
+import com.thejholmes.russound.RussoundTranslator;
+import com.thejholmes.russound.Zone;
+import com.thejholmes.russound.serial.SerialCommandReceiver;
+import com.thejholmes.russound.serial.SerialCommandSender;
+import org.openhab.binding.russoundserial.handler.RussoundSerialBridgeHandler;
+import org.openhab.binding.russoundserial.handler.RussoundSerialBridgeHandler.ZoneListener;
+import org.openhab.binding.russoundserial.handler.RussoundSerialZoneHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.russoundserial.RussoundSerialBindingConstants.SERIAL_BRIDGE;
+import static org.openhab.binding.russoundserial.RussoundSerialBindingConstants.ZONE_HANDLER;
+
+/**
+ * The {@link RussoundSerialHandlerFactory} is responsible for creating things and thing handlers.
+ *
+ * @author holmes - Initial contribution
+ */
+public class RussoundSerialHandlerFactory extends BaseThingHandlerFactory {
+  private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS =
+      Sets.newHashSet(SERIAL_BRIDGE, ZONE_HANDLER);
+
+  private final Logger logger = LoggerFactory.getLogger(RussoundSerialHandlerFactory.class);
+
+  @Override
+  public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+    return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+  }
+
+  @Override
+  protected ThingHandler createHandler(Thing thing) {
+    ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+    if (thingTypeUID.equals(SERIAL_BRIDGE)) {
+      String serialPort = (String) thing.getConfiguration().get("serialPort");
+      File deviceLocation = new File(serialPort);
+
+      try {
+        RussoundCommandSender commandSender = SerialCommandSender.Factory.fromFile(deviceLocation);
+        RussoundCommander commander =
+            Russound.Companion.sender("russound-serial-bridge", commandSender);
+
+        ZoneListener zoneInfoListener = new ZoneListener();
+        RussoundTranslator translator =
+            Russound.Companion.receiver("russound-serial-bridge", zoneInfoListener);
+        SerialCommandReceiver commandReceiver =
+            new SerialCommandReceiver(translator, new FileInputStream(deviceLocation));
+
+        return new RussoundSerialBridgeHandler((Bridge) thing, commander, commandReceiver,
+            zoneInfoListener);
+      } catch (FileNotFoundException e) {
+        logger.info("Couldn't find serialPort at {}", serialPort, e);
+        return null;
+      } catch (Exception e) {
+        logger.info("Unable to create RussoundSerialBridgeHandler", e);
```

Comment:
A user reading such an error message can probably not do anything about it, so please reduce the severity to `debug`.